docs/Private-IP-Configuration/README.md

```diff
@@ -0,0 +1,83 @@
+# Private IP Configuration Guide for Dataflow Pipeline Execution
+
+## Goals
+In this document, we’ll describe how to use Private IP for the execution of a dataflow pipeline.
+
+## Background
+When we are running the dataflow pipeline, GCP decides to spawn one or more new VM-instances. By default, each VM-instance will have an External IP address. 
+
+![VM Instance with External IP Address](https://user-images.githubusercontent.com/101806311/176452459-1f996a55-2604-4abc-a601-689442ba4d93.png?raw=true "VM Instance with External IP Address")
+
+Considering a billing account has a limited number of External IP addresses, we can skip this <u>overhead</u> by providing VPC-parameters as CLI-input of dataflow.
+
+Following table<font color="red">*</font> summarizes the required input parameters.
+
+| Field             | Description                                                                                                                                                              |
+|-------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| network           | The Compute Engine network for launching Compute Engine instances to run your pipeline. If not set, Google Cloud assumes that you intend to use a network named default. |
+| subnetwork        | The Compute Engine subnetwork for launching Compute Engine instances to run your pipeline.                                                                               |
+| no_use_public_ips | Command-line flag that sets use_public_ips to False. If the option is not explicitly enabled or disabled, the Dataflow workers use public IP addresses.                  |
+
+(<font color="red">*</font> excerpt from [GCP pipeline-options](https://cloud.google.com/dataflow/docs/reference/pipeline-options))
+
+## Steps to Configure VPC
+1. Configure VPC-Network & Subnetwork
+2. Configure Firewall-Rule
+3. Configure NAT & Router
+4. Sample commands to trigger dataflow pipeline execution using above options
+
+## Configure VPC-Network & Subnetwork
+1. Open GCP’s Create VPC-Network page.
+2. Provide name, description.
+3. Select “Subnet creation mode” as Custom.
+4. Provide name, description, region, IP address range & other
+5. Select “Private Google Access” as On.
+6. Complete VPC-Network creation by providing other required parameters. Refer to GCP’s [Create-VPC-Network](https://cloud.google.com/vpc/docs/create-modify-vpc-networks) documentation for more details.
+
+![VPC Network Details](https://user-images.githubusercontent.com/101806311/176452593-a8a45ba0-f6d3-46b6-b335-f3f0c34b6c31.png?raw=true "VPC Network Details")
```

Comment:
Optional, but recommended: In the images, can you circle the areas that need to be clicked / need user attention? It would be helpful to associate the steps in the instructions with how the page looks.

Reply:
@alxmrs In our document, we have put snaps of entities (i.e. VPC-Network, Sub-network, Firewall-Rule, NAT-Gateway & NAT-Router) post their successful creation. Hence there is no button or link in these images that we can highlight.